src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTagger.cs

```diff
@@ -19,56 +24,118 @@ internal sealed class InlineParameterNameHintsTagger : ITagger<IntraTextAdornmen
+        private readonly ForegroundThreadAffinitizedObject _threadAffinitizedObject;
 
-        public InlineParameterNameHintsTagger(ITextBuffer buffer, ITagAggregator<InlineParameterNameHintDataTag> tagAggregator)
+        public event EventHandler<SnapshotSpanEventArgs>? TagsChanged;
+
+        public InlineParameterNameHintsTagger(InlineParameterNameHintsTaggerProvider taggerProvider, ITextView textView, ITextBuffer buffer, ITagAggregator<InlineParameterNameHintDataTag> tagAggregator)
         {
+            _cache = new List<ITagSpan<IntraTextAdornmentTag>>();
+            _threadAffinitizedObject = new ForegroundThreadAffinitizedObject(taggerProvider.ThreadingContext);
+            _textView = textView;
             _buffer = buffer;
             _tagAggregator = tagAggregator;
+            _formatMap = taggerProvider.ClassificationFormatMapService.GetClassificationFormatMap(textView);
+            _hint = taggerProvider.ClassificationTypeRegistryService.GetClassificationType(InlineParameterNameHintsTag.TagId);
+            _formatMap.ClassificationFormatMappingChanged += this.OnClassificationFormatMappingChanged;
             _tagAggregator.TagsChanged += OnTagAggregatorTagsChanged;
         }
 
+        private void OnClassificationFormatMappingChanged(object sender, EventArgs e)
+        {
+            _threadAffinitizedObject.AssertIsForeground();
+            if (_format != null)
+            {
+                _format = null;
+                _cache.Clear();
+            }
+        }
+
         private void OnTagAggregatorTagsChanged(object sender, TagsChangedEventArgs e)
         {
+            _cache.Clear();
             var spans = e.Span.GetSpans(_buffer);
             foreach (var span in spans)
             {
                 TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(span));
             }
         }
 
+        private TextFormattingRunProperties Format
+        {
+            get
+            {
+                _threadAffinitizedObject.AssertIsForeground();
+                _format ??= _formatMap.GetTextProperties(_hint);
+                return _format;
+            }
+        }
+
         public IEnumerable<ITagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotSpanCollection spans)
         {
-            if (spans.Count <= 0)
+            if (spans.Count == 0)
             {
                 return Array.Empty<ITagSpan<IntraTextAdornmentTag>>();
             }
 
-            var tagsList = new List<ITagSpan<IntraTextAdornmentTag>>();
-            var dataTags = _tagAggregator.GetTags(spans);
             var snapshot = spans[0].Snapshot;
-            foreach (var tag in dataTags)
+            if (_cache.Count == 0 || snapshot != _cacheSnapshot)
+            {
+                // Calculate UI elements
+                _cache.Clear();
+                _cacheSnapshot = snapshot;
+
+                // Thye fullspan is really only the size of the textview plus 100 lines, not the entire file
```

Comment:
Might be worth clarifying this a bit further: the full span here is indeed the full span of the file (it makes this comment look stale). Might be better worth explaining that this is calling into the other tagger, and that tagger is only going to respond for the current active view, and will disregard the other requests for tags.